app/forms/verify_profile_form.rb

```diff
@@ -0,0 +1,54 @@
+class VerifyProfileForm
+  include ActiveModel::Model
+
+  validates :otp, presence: true
+  validate :validate_otp
+  validate :validate_pending_profile
+
+  attr_accessor :otp, :pii_attributes
+  attr_reader :user
+
+  def initialize(user:, otp: nil, pii_attributes: nil)
+    @user = user
+    @otp = otp
+    @pii_attributes = pii_attributes
+  end
+
+  def submit
+    if valid?
+      activate_profile
+      true
+    else
+      clear_fields
+      false
+    end
+  end
+
+  protected
+
+  def pending_profile
+    @_pending_profile ||= user.decorate.pending_profile
+  end
+
+  def validate_pending_profile
+    errors.add :base, :no_pending_profile unless pending_profile
+  end
+
+  def validate_otp
+    return if valid_otp?
+    errors.add :otp, :otp_incorrect
+  end
+
+  def valid_otp?
+    otp == pii_attributes.otp
```

Comment:
should this use `ActiveSupport::SecurityUtils.secure_compare` instead of the simple `==`?